index.js

```diff
@@ -194,7 +194,16 @@ module.exports = function inspect_(obj, options, depth, seen) {
     }
     if (typeof obj === 'object' && customInspect) {
         if (inspectSymbol && typeof obj[inspectSymbol] === 'function') {
-            return obj[inspectSymbol]();
+            return obj[inspectSymbol](maxDepth - depth, {
+                breakLength: 60,
+                colors: false,
+                customInspect: true,
+                depth: maxDepth,
+                maxArrayLength: 100,
+                seen: seen,
```

Comment:
passing this is a *very* bad idea.
```suggestion
```

Reply:
I am passing this because it is also passed by the nodejs inspect

Reply:
i'm pretty sure node's cache is different than ours, but either way, i'd rather not pass it here even if that causes breakage for a custom inspect function.

Reply:
Would it be better to pass an empty array? There is likely some code somewhere expect this to be an array?

Reply:
when i do `var o = { [util.inspect.custom]() { console.log(arguments); } }; util.inspect(o);` in the repl, this is what i see as the first argument:
```
{
    stylize: [Function: stylizeNoColor],
    showHidden: false,
    depth: 2,
    colors: false,
    customInspect: true,
    showProxy: false,
    maxArrayLength: 100,
    maxStringLength: 10000,
    breakLength: 80,
    compact: 3,
    sorted: false,
    getters: false,
    numericSeparator: false
  }
```
which doesn't contain a "seen" property.

I also note that there's a third argument, which is the original inspect function. should we pass that?

Reply:
The third argument is [a fancy new feature](https://github.com/nodejs/node/commit/4569674e3936853d21fb3af5f7e025c78363ab18) which, yeah, probably is a good idea to have.
`seen` has been passed in older versions of node, I couldn't locate the commit where this changed.